source/timesheet.js

```diff
@@ -19,23 +21,30 @@ class Timesheet{
         this.notes = timesheet.notes;
     }
 
-    save() {
+    async save() {
         return db.insertOne(this, collectionName);
     }
 
-    delete() {
+    async delete() {
+        await db.updateOne({_id : this.jobID}, { $pull: { timsheets : this.jobID} }, 'Jobs');
         return db.deleteOne({_id : this._id}, collectionName);
     }
 
-    validate() {
-        if(!(this._id instanceof mongodb.ObjectID)) throw new Error('timeSheetID is not valid');
-        if(!(this.date instanceof Date)) throw new Error('Date is not valid');
-        if(!(this.employeeID instanceof mongodb.ObjectID)) throw new Error('EmployeeID is not valid');
-        // if(typeof(this.employee) !== 'string') throw new Error('Street Name is not valid');
-        // if(typeof(this.city) !== 'string') throw new Error('City is not valid');
-        // if(typeof(this.state) !== 'string') throw new Error('State is not valid');
-        // if(typeof(this.zip) !== 'number') throw new Error('Zip is not valid');
-        // if(typeof(this.county) !== 'string') throw new Error('Country is not valid');
+    async validate() {
```

Comment:
validate should not be asynchronous because you're not calling any asynchronous code.

Reply:
fixed.